WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/BloggingRemindersViewModel.kt

```diff
@@ -45,21 +66,108 @@ class BloggingRemindersViewModel @Inject constructor(
-                Text(UiStringRes(R.string.set_your_blogging_goals_message)),
+        _selectedScreen.value = screen
+        launch {
+            bloggingRemindersStore.bloggingRemindersModel(siteId).collect {
+                _bloggingRemindersModel.value = it
+            }
+        }
+    }
+
+    private fun buildPrologue() = listOf(
+            Illustration(R.drawable.img_illustration_celebration_150dp),
+            Title(UiStringRes(R.string.set_your_blogging_goals_title)),
+            Text(UiStringRes(R.string.set_your_blogging_goals_message)),
+            PrimaryButton(
+                    UiStringRes(R.string.set_your_blogging_goals_button),
+                    enabled = true,
+                    ListItemInteraction.create(this::startDaySelection)
+            )
+    )
+
+    private fun buildSelection(bloggingRemindersModel: BloggingRemindersModel?): List<BloggingRemindersItem> {
+        // TODO Add selection view items
+        return listOf(
+                PrimaryButton(
+                        UiStringRes(R.string.blogging_reminders_notify_me),
+                        enabled = bloggingRemindersModel?.enabledDays?.isNotEmpty() == true,
+                        ListItemInteraction.create(bloggingRemindersModel, this::showEpilogue)
+                )
+        )
+    }
+
+    private fun buildEpilogue(): List<BloggingRemindersItem> {
+        // TODO Add epilogue view items
+        return listOf(
                 PrimaryButton(
-                        UiStringRes(R.string.set_your_blogging_goals_button),
-                        ListItemInteraction.create(this::onPrimaryClick)
+                        UiStringRes(R.string.blogging_reminders_done),
+                        enabled = true,
+                        ListItemInteraction.create(this::finish)
                 )
         )
     }
 
-    private fun onPrimaryClick() {
-        // TODO handle primary click
+    // TODO Call this method on day selection
+    fun selectDay(day: Day) {
+        val currentState = _bloggingRemindersModel.value!!
+        val enabledDays = currentState.enabledDays.toMutableSet()
+        if (enabledDays.contains(day)) {
+            enabledDays.remove(day)
+        } else {
+            enabledDays.add(day)
+        }
+        _bloggingRemindersModel.value = currentState.copy(enabledDays = enabledDays)
+    }
+
+    private fun startDaySelection() {
+        _selectedScreen.value = SELECTION
+    }
+
+    private fun showEpilogue(bloggingRemindersModel: BloggingRemindersModel?) {
+        if (bloggingRemindersModel != null) {
+            bloggingRemindersStore.updateBloggingReminders(bloggingRemindersModel)
+            // TODO Add logic to save state and schedule notifications here
+            _selectedScreen.value = EPILOGUE
+        }
+    }
+
+    private fun finish() {
+        _isBottomSheetShowing.value = Event(false)
+    }
+
+    fun saveState(outState: Bundle) {
```

Comment:
Could we use [SavedStateHandle](https://developer.android.com/topic/libraries/architecture/viewmodel-savedstate) here!